### PR TITLE
PR: Update with latest palantir changes

### PR DESCRIPTION
**Pull request by goanpeca**
Fixes https://github.com/spyder-ide/spyder-pyls/issues/7

### CODE REVIEW

File: pyls/plugins/jedi_completion.py [hunk 1]
```diff
@@ -8,14 +8,57 @@
...
+    'constant': lsp.CompletionItemKind.Variable,
+    'variable': lsp.CompletionItemKind.Variable,
+    'value': lsp.CompletionItemKind.Value,
+    'param': lsp.CompletionItemKind.Variable,
+    'statement': lsp.CompletionItemKind.Keyword,
+}
+
 
 @hookimpl
-def pyls_completions(document, position, config=None, workspace=None):
```
Comment (ccordoba12) [inline pyls/plugins/jedi_completion.py:13]: This change breaks what @steff456 did in PR #5. So you need to preserve the same signature as the Palantir (to maintain compatibility with upstream) and add workspace at the end:

    pyls_completions(config, document, position, workspace=None):

You also need to make the `pyls_completions` in `rope_completion` to have the same signature:

https://github.com/spyder-ide/spyder-pyls/blob/develop/pyls/plugins/rope_completion.py#L18

**Review by ccordoba12 — COMMENTED** [review]
I left one minor comment. Please also take a deep look at PR #5 to see what other upstream changes could break our code.

File: pyls/plugins/jedi_completion.py (hunk 1 above)
Comment (ccordoba12) [reply pyls/plugins/jedi_completion.py:13]: I meant the same signature as this function, to make the race between Rope and Jedi work as expected.

Comment (goanpeca) [reply pyls/plugins/jedi_completion.py:13]: Fixed this and the tests. I think it should be fine now @ccordoba12

Comment (ccordoba12) [reply pyls/plugins/jedi_completion.py:13]: Yep, thanks for this :+1:

**Review by ccordoba12 — APPROVED** [review]
Thanks @goanpeca, looks good to me now.

**Pull request merged**